### ISSUE TITLE
Tests failing on deployed(in-use) CRM

### ISSUE THREAD

**Issue by adamjakab** [salesagility/SuiteCRM#1932]
When I run tests on a deployment which is already in use(customizations made to fields through studio or to which tabs to show in menu) tests are failing.
#### Issue

Hard coded expected arrays like [this](https://github.com/salesagility/SuiteCRM/blob/master/tests/tests/include/utils/securityUtilsTest.php#L12) when tested against tabs that user can see will only pass if user does not change tabs visibility in his profile.
#### Expected Behavior

Test should pass even if user does change tabs visibility.
#### Actual Behavior

 As soon as user does change which tabs should be visualized it will break the test.
#### Possible Fix

Get rid of hard coded array, get array of all available modules from $beanList, exclude modules that user does not have access to (with get_user_allowed_modules($user->id)) then loop through array provided by TabController::get_tabs() and only fail if it has module that should not be allowed for user.
#### Steps to Reproduce
1. Install
2. make modifications to admin user's visible tabs (or add custom module)
3. run tests
#### Context

Why? Because When we make modifications/fixes to CRM we need a working copy of the CRM with personalizations made - it is not enough to test a clean install.
### Similar issues

This type of issue is also true for:
1.  **queries**: as soon as I add a custom field in any of the modules, the testcreate_export_query tests will fail because the expected sql statement is hard coded. These tests are really ugly not only because they cannot consider custom fields but also because they do a string comparison between 2 sql statements so even a possible whitespace difference between them will make the tests fail.
2. **language**: Tests do not use translations - If I happen to set the default language to something different from 'en_us' - I will get error failing because translatable strings are hard coded in the tests:
`Failed asserting that 'Creazione Rapida&nbsp;<a id='' onclick='return quick_create_overlib("", "SuiteR", this);' href="#" ><img src="themes/SuiteR/images/advanced_search.gif?v=G1JzxNj_0FZs1WnHLWz4jA"    border='0' align='absmiddle' alt="Creazione Rapida" /></a>' matches PCRE pattern "~Quick Create&nbsp;\<a id\='' onclick\='return quick_create_overlib\("", "SuiteR", this\);' href\="#" \>\<img src\="themes/SuiteR/images/advanced_search\.gif\?v\=[\w-]+"    border\='0' align\='absmiddle' alt\="Creazione Rapida" /\>\</a\>~".`
#### Environment
- SuiteCRM: 7.1.1
- Php: 7.0
- Mysql: 5.5
- Apache: 2.4.7
- Browser: Google Chrome v. 52.0.2743.116 (64-bit)
- OS: Ubuntu 14.04 LTS

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Tests failing on deployed(in-use) CRM](https://trello.com/c/L4NajITu/81-tests-failing-on-deployed-in-use-crm)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
